### PR TITLE
Delete old serialization extension points and fallbacks

### LINKED ISSUE #47223: extract-and-log-one should become extract-one

### PR DESCRIPTION
**Pull request by crisptrutski**
Closes https://github.com/metabase/metabase/issues/47223

I've taken some liberties with the instructions, following the spirit:

- Require specs
- Remove defmethod overrides
- Remove fallbacks
- Inline where natural

### CODE REVIEW

**Comment by crisptrutski** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/metabase/metabase/47273?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#47273** <a href="https://app.graphite.dev/github/pr/metabase/metabase/47273?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#47268** <a href="https://app.graphite.dev/github/pr/metabase/metabase/47268?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#47240** <a href="https://app.graphite.dev/github/pr/metabase/metabase/47240?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @crisptrutski and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/metabase/models/serialization.clj [hunk 1]
```diff
@@ -314,39 +312,14 @@
 
   `:copy` and `:skip` are vectors of field names. `:skip` is only used in tests to check that all fields were
   mentioned.
-
-  `:transform` is a map from field name to a `{:ser (fn [v] ...) :des (fn [v] ...)}` map with functions to
+  j
   serialize/deserialize data.
 
-  For behavior, see `extract-by-spec` and `xform-by-spec`."
+  For behavior, see `extract-one` and `xform-by-spec`."
```
Comment (crisptrutski) [inline src/metabase/models/serialization.clj:318]: Lack of symmetry here sucks. Hmm perhaps I prefer keeping `by-spec`.

Comment (piranha) [reply src/metabase/models/serialization.clj:318]: or maybe `xform-one`? :)

**Review by sansolovyov — APPROVED** [review]
That's nice, but I also wish for more consistent naming.

I'm also thinking if a follow-up PR makes sense to remove dynamic bindings (i.e. changing import/export signature to be like `[value key instance]`). I did not do that because `*import-fk*` and the like were used in lots of places, but now it seems more doable/logical?

**Pull request merged**